[RyanQiu1] iP

RyanQiu1 opened the pull request:
# Duke Bot
> Bot that is here to help you and your tasks out!

Duke bot is: 
- Texted Based
- Easy to use
- ~~Slow~~ Fast!

Here are the steps involved!
1. Download the Jar file from [here](https://github.com/RyanQiu1/ip/releases/tag/A-Jar)
2. Click It!
3. Add your tasks!
4. Use it daily! :grinning:

Most importantly **Free** Software

Features in this software
- [ ] Add todo tasks, events and deadlines
- [ ] Delete tasks whenever you want
- [ ] Saves your tasks and updates automatically!

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

RyanQiu1 committed:
Greet, Echo, Exit functions added

RyanQiu1 committed:
Add, List and new StorageList class to the functions

RyanQiu1 committed:
Add the ability to mark tasks as done and undone

RyanQiu1 committed:
Add support for tracking three types of tasks, todos, deadline, events

RyanQiu1 committed:
Automated Text UI Testing function added

RyanQiu1 committed:
Making Duke to deal with errors such as incorrect inputs entered by the user, first is to handle the unknown command error and next is to handle the empty field error

RyanQiu1 committed:
Add support for deleting tasks from the list.

RyanQiu1 committed:
Added enums to my classes

RyanQiu1 committed:
Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when Duke starts up.

RyanQiu1 committed:
Teach Duke how to understand dates and times, added these functions

RyanQiu1 committed:
Made code more OOP styled and according to the standards

RyanQiu1 committed:
Organized the classes into suitable java packages, Command, Task, etc

RyanQiu1 committed:
Update File allocation of exception

RyanQiu1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

RyanQiu1 committed:
Use Gradle to automate some of the build tasks of the project, add 2 tests to the classes

RyanQiu1 committed:
Added JavaDoc comments to the code.

RyanQiu1 committed:
Tweaked the code to comply with a given coding standard

RyanQiu1 committed:
Allow users to find a task by searching for a keyword from the chatbot

RyanQiu1 committed:
Merge tag 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java

RyanQiu1 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/parser/Parser.java

RyanQiu1 committed:
Update Bye Command

RyanQiu1 committed:
Resolve all the conflict errors

RyanQiu1 committed:
Added Checkstyle to detect code violations

joyngjr commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,22 @@
+        this.listNum = taskNumberMark;
+    }
+
+    public boolean execute(TaskList tasks, Ui ui, StorageList storage) {
```
Perhaps this boolean method can be made to sound like a boolean (eg. canExecute)?

joyngjr commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,35 @@
+        this.to = checkerTo[1];
+    }
+
+    public boolean execute(TaskList tasks, Ui ui, StorageList storage) {
```
I noticed your naming convention for boolean methods in several other places too. Might be good to make them sound more like a boolean.

joyngjr commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,64 @@
+        String[] input2 = input.split(" ");
+        Type t = Type.valueOf(input2[0].toLowerCase());
+        switch (t) {
+            case todo:
```
Should the indentation of your switch-case statement be the same?

joyngjr commented on src/main/java/duke/storage/StorageList.java:
```diff
@@ -0,0 +1,72 @@
+    public void updateStorage() {
+        try {
+            File dir = new File("data");
+            if (!dir.exists()) dir.mkdirs();
```
Shouldn't your conditionals be placed on a different line, and be wrapped in curly braces?

joyngjr commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,71 @@
+
+    public String toString() {
+        if (typeofTask.equals(""))
+            return "[" + this.getStatusIcon() + "]" + " " + this.description;
```
Same issue as with StorageList.java, shouldn't the if-block be wrapped in curly braces?

RyanQiu1 replied:
Yes u are right, i have rectified it.

joyngjr commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+    public ArrayList<Task> find(String message) {
+        ArrayList<Task> arrStr = new ArrayList<>();
+        for (Task t: list) {
+            if (t.toString().contains(message)) {
```
Good job adding curly braces here! 👍

joyngjr reviewed (commented):
Looks good!
Most of your code follows the coding standards, just that some minor tweaks are needed for your boolean methods, switch-case indentation and if-else blocks.

RyanQiu1 committed:
Used JavaFX to make a GUI

RyanQiu1 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	build.gradle
#	src/main/java/duke/Duke.java
#	src/main/java/duke/command/ByeCommand.java
#	src/main/java/duke/command/DeadlineCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/EventCommand.java
#	src/main/java/duke/command/FindCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/TodoCommand.java
#	src/main/java/duke/command/UnmarkCommand.java
#	src/main/java/duke/task/TaskList.java

RyanQiu1 committed:
Fixed build gradle file error

RyanQiu1 committed:
Fixed some code formatting error.

RyanQiu1 committed:
Default duke text and stage default image.

There is a addition of a title and appears more aesthetically pleasing.

Adding the images and improving the UI, with setTitle() and getIcons().add().

To enable the user to see the application they are accessing and the icon of the software.

RyanQiu1 committed:
Added assertions and fixed the removed unused UI

RyanQiu1 committed:
Improve code quality by changing names, changing oop structure.

RyanQiu1 committed:
Merge pull request #1 from RyanQiu1/branch-A-Assertions

Added assertions and fixed the removed unused UI

RyanQiu1 committed:
Merge branch 'master' into branch-A-CodeQuality

RyanQiu1 committed:
Merge pull request #2 from RyanQiu1/branch-A-CodeQuality

Improve code quality by changing names, changing oop structure.

RyanQiu1 committed:
Merge branch 'master' of https://github.com/RyanQiu1/ip into branch-A-CodeQuality

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/command/DeadlineCommand.java
#	src/main/java/duke/command/FindCommand.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/TodoCommand.java
#	src/main/java/duke/command/UnmarkCommand.java

RyanQiu1 committed:
Used GitHub Actions to set up Continuous Integration (CI).

RyanQiu1 committed:
Fixed Checkstyle error.

wxxedu commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,34 @@
+    private String timing;
+
+    public DeadlineCommand(String fullCommand) {
+        String[] checker = fullCommand.split("/by ");
```
May I know why this is called `checker`???

wxxedu commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,22 @@
+ * Delete Command to remove the tasks users do not want, according to its number on the list.
+ */
+public class DeleteCommand extends Command {
+    private int listNum;
```
Can I clarify what `listNum` is used for? Would it be better to call it `index` or something?

wxxedu commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,35 @@
+    public EventCommand(String input) {
+        String[] checkerSlash = input.split("/");
+        String[] checkerEvent = checkerSlash[0].split("event ");
+        String[] checkerFrom = checkerSlash[1].split("from ");
```
Would it be better if we rename it to `fromChecker` or something related?

wxxedu reviewed (commented):
Overall, I think your code is really structured and follows the coding standard very well.

RyanQiu1 committed:
Adjusted gradle.yml file to fit CI constraints.

RyanQiu1 committed:
Implemented the tagging of tasks with a few checkstyle fixes.

RyanQiu1 committed:
Fixed the Assertion test.

RyanQiu1 committed:
Update for the name ok chatbot and the fonts.

RyanQiu1 committed:
Updated the user guide.

RyanQiu1 committed:
Fixed Checkstyle issues.

RyanQiu1 committed:
Update Ui screenshots and readme to follow the required format.

RyanQiu1 committed:
Update readme file to fix hyperlink errors.

RyanQiu1 committed:
Update readme for relative hyperlink.

RyanQiu1 committed:
Update readme to fixed relative hyperlink issue.

RyanQiu1 committed:
Update Junit Tests to include more test cases.

RyanQiu1 committed:
Fix checkstyle issues.

RyanQiu1 committed:
Fix readme table rendering.

RyanQiu1 committed:
Fix readme table error.

RyanQiu1 committed:
Fix readme file.

RyanQiu1 committed:
Fix readme file.

RyanQiu1 committed:
Edit starting message of bot.
